File: code/python/2021/sept/intersection-of-two-arrays-ii.py
from typing import Collection
import collections


class Solution:
    def intersect(self, nums1, nums2):
        C = Collection.Counter
        return list((C(nums1) & C(nums2)).elements())

    def intersect(self, nums1, nums2):
        return list(
            (collections.Counter(nums1) & collections.Counter(nums2)).elements()
        )

File: archive/go/2021/sept/intersection-of-two-arrays-ii.go
package sept

func intersect(nums1 []int, nums2 []int) []int {
	set := make(map[int]int)
	var res []int
	for _, i := range nums1 {
		set[i]++
	}
	for _, i := range nums2 {
		if v, ok := set[i]; ok {
			if v > 0 {
				res = append(res, i)
				set[i]--
			}
		}
	}
	return res
}
